core/src/main/scala/org/http4s/rho/bits/PathTree.scala

```diff
@@ -58,14 +58,18 @@ private[rho] object PathTree {
   def splitPath(path: String): List[String] = {
     val buff = new ListBuffer[String]
     val len = path.length
+
     @tailrec
-    def go(i: Int, begin: Int): Unit = if (i < len) {
-      if (path.charAt(i) == '/') {
-        if (i > begin) buff += path.substring(begin, i)
-        go(i + 1, i + 1)
-      } else go(i + 1, begin)
-    } else {
-      buff += path.substring(begin, i)
+    def go(i: Int, begin: Int): Unit = {
+      def buffer(): Unit = buff += java.net.URLDecoder.decode(path.substring(begin, i), "UTF-8")
```

Comment:
I think in the path we don't want to decode `+` as `space` which java.net.URLDecoder does. You could use the http4s `UrlCoding.decode` utility function. It's http4s private, but since rho lives in the http4s package it should work, although I'm not thrilled about it :unamused:. If we hadn't been so lucky, an alternative solution is to replace any '+' chars with "%2b" before doing the decoding (preferably before doing the slicing so its all done in one pass).

I'd prefer to inline `buffer` since its only used twice, but if you feel strongly about it, then its fine.